Mathlib/NumberTheory/ModularForms/SlashInvariantForms.lean

```diff
@@ -215,4 +215,31 @@ instance (Γ : Subgroup SL(2, ℤ)) : IntCast (SlashInvariantForm Γ 0) where
 @[simp, norm_cast]
 theorem coe_intCast (z : ℤ) : ⇑(z : SlashInvariantForm Γ 0) = z := rfl
 
+/-- Translating a `SlashInvariantForm` by `SL(2, ℤ)`, to obtain a new `SlashInvariantForm`. -/
+noncomputable def translateGLPos [SlashInvariantFormClass F Γ k] (f : F) (g : GL(2, ℝ)⁺) :
+    SlashInvariantForm (ModularGroup.conjGLPos Γ g) k where
+  toFun := f ∣[k] g
+  slash_action_eq' := fun j hj ↦ by
+    obtain ⟨y, hy, hy'⟩ := ModularGroup.mem_conjGLPos.mp hj
+    simp only [ModularForm.SL_slash, ← hy', ← SlashAction.slash_mul, mul_assoc,
+      mul_inv_cancel_left]
+    rw [SlashAction.slash_mul, ← ModularForm.SL_slash,
+      SlashInvariantFormClass.slash_action_eq f _ hy]
+
+@[simp]
+lemma coe_translateGLPos [SlashInvariantFormClass F Γ k] (f : F) (g : GL(2, ℝ)⁺) :
+    translateGLPos f g = ⇑f ∣[k] g :=
+  rfl
+
+/-- Translating a `SlashInvariantForm` by `SL(2, ℤ)`, to obtain a new `SlashInvariantForm`. -/
+noncomputable def translate [SlashInvariantFormClass F Γ k]
+    (f : F) (g : SL(2, ℤ)) : SlashInvariantForm (Γ.map <| MulAut.conj g⁻¹) k where
```

Comment:
For what it's worth, I think `Γ.map <| MulAut.conj g⁻¹` can also be spelt with `ConjAct`.